build_tools/install_rocm_from_artifacts.py

```diff
@@ -77,81 +81,44 @@ def _create_output_directory(args):
     log(f"Created directory {output_dir_path}")
 
 
-def _get_github_release_assets(release_tag, amdgpu_family, release_version):
+def _retrieve_s3_release_assets(
+    release_bucket, amdgpu_family, release_version, output_dir
+):
     """
     Makes an API call to retrieve the release's assets, then retrieves the asset matching the amdgpu family
     """
-    github_release_url = (
-        f"https://api.github.com/repos/ROCm/TheRock/releases/tags/{release_tag}"
-    )
-    headers = {
-        "Accept": "application/vnd.github+json",
-        "X-GitHub-Api-Version": "2022-11-28",
-    }
-    # If GITHUB_TOKEN environment variable is available, include it in the API request to avoid a lower rate limit
-    gh_token = os.getenv("GITHUB_TOKEN", "")
-    if gh_token:
-        headers["Authentication"] = f"Bearer {gh_token}"
-
-    request = Request(github_release_url, headers=headers)
-    with urlopen(request) as response:
-        if response.status == 403:
+    # TODO: add logic for latest
+
+    asset_name = f"therock-dist-{PLATFORM}-{amdgpu_family}-{release_version}.tar.gz"
+    # URL encoding the asset_name, in cases of "+" symbols
+    encoded_asset_name = urllib.parse.quote_plus(asset_name)
+    s3_release_url = f"https://{release_bucket}.s3.amazonaws.com/{encoded_asset_name}"
```

Comment:
It would be sufficient to just replace `+` characters with `%2B`, whereas `quote_plus` replaces `/` with `%2F` and quotes `+` characters.
```suggestion
    asset_name = f"therock-dist-{PLATFORM}-{amdgpu_family}-{release_version}.tar.gz".replace("+", "%2B")
    s3_release_url = f"https://{release_bucket}.s3.amazonaws.com/{asset_name}"
```
If you want to stick with `urllib`
```suggestion
    asset_name = f"therock-dist-{PLATFORM}-{amdgpu_family}-{release_version}.tar.gz"
    # URL encoding the asset_name, in cases of "+" symbols
    encoded_asset_name = urllib.parse.quote(asset_name)
    s3_release_url = f"https://{release_bucket}.s3.amazonaws.com/{encoded_asset_name}"
```
might do the job as well, as we don't have spaces in the asset names.